docs/features/deploy.md

```diff
@@ -0,0 +1,127 @@
+
+```sh
+$ algokit deploy [OPTIONS] [ENVIRONMENT_NAME]
+```
+
+This command deploys smart contracts from an AlgoKit compliant repository to the specified network.
+
+### Options
+
+- `--command, -C TEXT`: Specifies a custom deploy command. If this option is not provided, the deploy command will be loaded from the `.algokit.toml` file.
+- `--interactive / --non-interactive, --ci`: Enables or disables the interactive prompt for mnemonics. When the CI environment variable is set, it defaults to non-interactive.
+- `--path, -P DIRECTORY`: Specifies the project directory. If not provided, the current working directory will be used.
+- `-h, --help`: Show this message and exit.
+
+## Environment files
+
+AlgoKit `deploy` employs both a general and network-specific environment file strategy. This allows you to set environment variables that are applicable across all networks and others that are specific to a given network.
+
+The general environment file (`.env`) should be placed at the root of your project. This file will be used to load environment variables that are common across deployments to all networks.
+
+For each network you're deploying to, you can optionally have a corresponding `.env.[network_name]` file. This file should contain environment variables specific to that network. Network-specific environment variables take precedence over general environment variables.
+
+The directory layout would look like this:
+
+```md
+.
+├── ... (your project files and directories)
+├── .algokit.toml # Configuration file for AlgoKit
+├── .env # (OPTIONAL) General environment variables common across all deployments
+└── .env.[{mainnet|testnet|localnet|betanet|custom}] # (OPTIONAL) Environment variables specific to deployments to a network
+```
+
+> ⚠️ Please note that creating `.env` and `.env.[network_name]` files is only necessary if you're deploying to a custom network or if you want to override the default network configurations provided by AlgoKit. AlgoKit comes with predefined configurations for popular networks like `TestNet`, `MainNet`, `BetaNet`, or AlgoKit's `LocalNet`.
+
+The logic for loading environment variables is as follows:
+
+- If a `.env` file exists, the environment variables contained in it are loaded first.
+- If a `.env.[network_name]` file exists, the environment variables in it are loaded, overriding any previously loaded values from the `.env` file for the same variables.
+
+## AlgoKit Configuration File
+
+AlgoKit uses a configuration file called `.algokit.toml` in the root of your project. The configuration file can be created using the `algokit init` command. This file will define the deployment commands for the various network environments that you want to target.
+
+Here's an example of what the `.algokit.toml` file might look like. The example below will always prompt for the deployer mnemonic secret (unless specified in env variables) for any network, except for the localnet, where it will not prompt for any secrets.
```

Comment:
```suggestion
Here's an example of what the `.algokit.toml` file might look like. When deploying it will prompt for the `DEPLOYER_MNEMONIC` secret unless it is already defined as an environment variable or is deploying to localnet.
```